### PR TITLE
Added Physics Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this mete ticket https://github.com/ignitionrobotics/sdformat/issues/931

## Summary

Added Physics Python interface

## Test it

```python
from sdformat import Physics

physics = Physics()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pyWorld.cc [hunk 1]
```diff
@@ -163,6 +163,13 @@ void defineWorld(pybind11::object module)
           "Set the atmosphere model associated with this world.")
      .def("physics_count", &sdf::World::PhysicsCount,
           "Get the number of physics profiles.")
+     .def("physics_by_index",
+          pybind11::overload_cast<uint64_t>(
+            &sdf::World::PhysicsByIndex),
+          pybind11::return_value_policy::reference_internal,
+          "Get a mutable physics profile based on an index.")
+     .def("physics_name_exists", &sdf::World::PhysicsNameExists,
```
Comment (azeey) [inline python/src/sdf/pyWorld.cc:171]: Binding from `World::PhysicsDefault` is missing

**Review by azeey — CHANGES_REQUESTED** [review]
Looks good. Missing one member function, I think.

File: python/src/sdf/pyWorld.cc (hunk 1 above)
Comment (ahcorde) [reply python/src/sdf/pyWorld.cc:183]: https://github.com/gazebosim/sdformat/pull/995/commits/303aa17e5bffb05a5fe8a79456e84470aad866fe

File: python/test/pyWorld_TEST.py [hunk 2]
```diff
@@ -163,27 +163,44 @@ def test_set(self):
...
-#
-#   self.assertNotEqual(None, world.Scene())
-#   self.assertEqual(Color::Blue, world.Scene().Ambient())
-#   self.assertEqual(Color::Red, world.Scene().Background())
-#   self.assertTrue(world.Scene().Grid())
-#   self.assertTrue(world.Scene().Shadows())
-#   self.assertTrue(world.Scene().OriginVisual())
-# }
+
+    def test_set_scene(self):
```
Comment (azeey) [inline python/test/pyWorld_TEST.py:167]:
```suggestion
    def test_set_physics(self):
```

**Review by azeey — APPROVED** [review]
LGTM! Just one minor comment

**Comment by azeey** [comment]
`pyWorld_TEST` is failing. And it looks like Jenkins is not capturing test failures from python tests. There are a bunch of failures on homebrew, but the test results shows 0 failures. Is this a known issue?

**Pull request merged**